static/app/actionCreators/navigation.tsx

```diff
@@ -19,6 +20,12 @@ export function navigateTo(
 
   const projectById = ProjectsStore.getById(comingFromProjectId);
 
+  const onFinish = useCallback((modalProps: ModalRenderProps, path: string) => {
+    return Promise.resolve(modalProps.closeModal()).then(() => {
+      router.push(path);
+    });
```

Comment:
Is this more confusing than what we had before? to me it kind of is 😬

Reply:
Tbh, I kind of just want to delete this because deferring event loop to get store updates is 🤮. I'll revert, but I'll try and find a better way

Reply:
I've reverted. The downside is that we do leak the timeout now, but it's arguably equally annoying. We should never have side effects like this in the code where we cannot tap into that callback. I'm tempted to change onFinish signature here so that it accepts the setTimeoutId...

Reply:
Actually, that is probably the thing to do